src/garage/torch/modules/gaussian_mlp_module.py

```diff
@@ -0,0 +1,222 @@
+
+            if adaptive_std:
+                self.adaptive_std_module = self._get_adaptive_std_network()
+
+    def _get_shared_mean_std_network(self):
+        return MLPModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim * 2,
+            hidden_sizes=self._hidden_sizes,
+            hidden_nonlinearity=self._hidden_nonlinearity,
+            hidden_w_init=self._hidden_w_init,
+            hidden_b_init=self._hidden_b_init,
+            output_nonlinearity=self._output_nonlinearity,
+            output_w_init=self._output_w_init,
+            output_b_init=self._init_shared_b,
+            layer_normalization=self._layer_normalization
+        )
+
+    def _init_shared_b(self, b):
+        nn.init.zeros_(b[..., :self._action_dim])
+        nn.init.constant_(b[..., self._action_dim:], self._init_std_param.item())
+
+    def _get_mean_network(self):
+        return MLPModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=self._hidden_sizes,
+            hidden_nonlinearity=self._hidden_nonlinearity,
+            hidden_w_init=self._hidden_w_init,
+            hidden_b_init=self._hidden_b_init,
+            output_nonlinearity=self._output_nonlinearity,
+            output_w_init=self._output_w_init,
+            output_b_init=self._output_b_init,
+            layer_normalization=self._layer_normalization)
+
+    def _get_adaptive_std_network(self):
+        return MLPModule(
+            input_dim=self._input_dim,
+            output_dim=self._action_dim,
+            hidden_sizes=self._std_hidden_sizes,
+            hidden_nonlinearity=self._std_hidden_nonlinearity,
+            hidden_w_init=self._std_hidden_w_init,
+            hidden_b_init=self._std_hidden_b_init,
+            output_nonlinearity=self._std_output_nonlinearity,
+            output_w_init=self._std_output_w_init,
+            output_b_init=self._init_adaptive_b,
+            layer_normalization=self._layer_normalization)
+
+    def _init_adaptive_b(self, b):
+        return nn.init.constant_(b, self._init_std_param.item())
+
+    def forward(self, inputs):
+        if self._std_share_network:
+            output = self.shared_mean_std_network(inputs)
+            mean = output[..., :self._action_dim]
+            std = output[..., self._action_dim:]
+        else:
+            mean = self.mean_module(inputs)
+
+            if self._adaptive_std:
+                std = self.adaptive_std_module(inputs)
+            else:
+                broadcast_shape = list(inputs.shape[:-1]) + [self._action_dim]
+                x = torch.zeros(*broadcast_shape)
+                std = x + self._init_std_param
+
+        if self._min_std_param or self._max_std_param:
+            std = std.clamp(min=to_scalar(self._min_std_param), max=to_scalar(self._max_std_param))
+
+        if self._std_parameterization == 'exp':
+            log_std_var = std
```

Comment:
Shouldn't this be `std.log()`?

Reply:
I followed tensorflow [GaussianMLPModule](https://github.com/rlworkgroup/garage/blob/3bf0b2b40c3db30c66f95460f0280b442457cc7c/src/garage/tf/models/gaussian_mlp_model.py#L215).
but I think you are right.. could you check the tensorflow version too?

Reply:
I think there is a difference here. They assume that the neural network output is always log std but here your output is a std. I think we should also do that too. But to make the naming more clear: name the output of neural network as `log_std` and name the after-parameterized variable as `std_param`. We should just change to use `std_param` everywhere since it does not have to be a log std.

The naming in the tf tree is a bit confusing. @nish21 @ahtsan, I suggest fix this in the future?

Reply:
Oh sorry, I made a mistake. Didn't notice that the second log in the softplus...

@yonghyuc ok the naming is fine. Just change the naming of model output to log_std..

Reply:
you mean change the name "log_std_var" to "log_std"?

Reply:
std -> log_std

Reply:
and.. if the output from std_network is log_std..
in the softplus part, isn't the equation should be` log ( log ( 1 + e^ (e^x) ) )`?
because x is log_std not std so do we need to exponentiate the x?

Reply:
Talked with Zhanpeng and decided to change the code into `log ( log ( 1 + e^ (e^x) ) )`